src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,117 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_tracer.h"
+#include <yaml-cpp/yaml.h>
+
+namespace tracing {
+
+  const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+  TracerConfiguration tracing::Tracer::tracer_conf;
```

Comment:
you have this global `TracerConfiguration` here, and another one in `class Tracer` as a static member variable

since `TracerConfiguration` is a config observer, i don't think it's safe to rely on static destruction; by the time its destructor runs and calls `cct->_conf.remove_observer(this)`, the `CephContext* cct` will have already been destroyed

Reply:
it's the same object. the one inside the class it's the definition.

so since the `CephContext` may be destroyed before, we just don't need this d'tor.

Reply:
btw, why do we have the conf as a static member?
probably better and more "future proof" to make it a regular member

Reply:
conf is a static member because it is used to disable all tracers.
I don't think we need a conf per tracer (per thread)

Reply:
ok. then probably it would be better if we pass in pointer to a conf, which exists outside of the Tracer.
making it static really impose the specific way we want to use it on the Tracer object.

Reply:
i'm not sure we need a config observer for this, when we can just check `if (cct->conf->rgw_jaeger_enabled)` when starting a span